src/PIL/Image.py

```diff
@@ -2184,12 +2225,13 @@ def thumbnail(self, size, resample=BICUBIC):
         if size == self.size:
             return
 
-        res = self.draft(None, size)
-        if res is not None:
-            box = res[1]
+        if max_reduce is not None:
+            res = self.draft(None, (size[0] * max_reduce, size[1] * max_reduce))
+            if res is not None:
+                box = res[1]
 
         if self.size != size:
-            im = self.resize(size, resample, box=box)
+            im = self.resize(size, resample, box=box, max_reduce=max_reduce)
```

Comment:
On the other hand, `thumbnail()` takes `max_reduce` into account when working with any other images and achieves compared speed as when working with JPEGs.

Huge win!